minetest/minetest
# [no sq] Texture/rendering fixes and improvements

followup to #15396 

context for commit 2 is [here](https://learnopengl.com/Advanced-OpenGL/Depth-testing):

> Another great trick at the cost of some performance is to use a higher precision depth buffer. Most depth buffers have a precision of 24 bits, but most GPUs nowadays support 32 bit depth buffers, increasing the precision by a significant amount. So _at the cost of some performance_ you'll get much more precision with depth testing, reducing z-fighting. 

## To do

This PR is Ready for Review.


## How to test

first commit:
- apply
```diff
diff --git a/src/irrlicht_changes/CGUITTFont.h b/src/irrlicht_changes/CGUITTFont.h
--- a/src/irrlicht_changes/CGUITTFont.h
+++ b/src/irrlicht_changes/CGUITTFont.h
@@ -143,7 +143,7 @@ namespace gui
 				bool flgmip = driver->getTextureCreationFlag(video::ETCF_CREATE_MIP_MAPS);
 				driver->setTextureCreationFlag(video::ETCF_CREATE_MIP_MAPS, false);
 				bool flgcpy = driver->getTextureCreationFlag(video::ETCF_ALLOW_MEMORY_COPY);
-				driver->setTextureCreationFlag(video::ETCF_ALLOW_MEMORY_COPY, true);
+				driver->setTextureCreationFlag(video::ETCF_ALLOW_MEMORY_COPY, false);
 
 				// Set the texture color format.
 				switch (pixel_mode)
@@ -176,6 +176,8 @@ namespace gui
 			{
 				if (!dirty) return;
 
+				fprintf(stderr, "updating glyph texture\n");
+
 				void* ptr = texture->lock();
 				video::ECOLOR_FORMAT format = texture->getColorFormat();
 				core::dimension2du size = texture->getOriginalSize();
```
- ensure that "updating glyph texture" shows up *and* that the font rendering works fine

other commits:
check that everything still works, in particular:
* sun/moon tonemapping
* minimap
* dynamic shadows
